File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/player/action/AttackAction.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.action;

import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Player;
import edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.player.Ship;

/**
 * Represents player attack action.
 */
public class AttackAction implements PlayerEncounterAction {

    @Override
    public String doAction(Player player, Ship otherShip, EncounterState es) {
        String retString = "";

        //attack ship
        Ship playerShip = player.getShip();

        retString += "You attack the other ship.\n";
        boolean otherShipDead = player.attackShip(otherShip);
        if (otherShipDead) {
            retString += "The other ship explodes.\n";
            es.conclude();
            return retString;
        }

        retString += "The other ship attacks you.\n";
        boolean playerShipDead = otherShip.attackShip(playerShip);
        if (playerShipDead) {
            retString += "Your ship explodes. You died!\n";
            es.conclude();
            return retString;
        }

        return retString;
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/player/action/AttackAction.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

class AttackAction : PlayerEncounterAction {
    
    func doAction(player: Player, otherShip: Ship, es: EncounterState) -> String {
        var retString = ""
        
        let playerShip = player.ship!
        
        retString += "You attack the other ship.\n"
        let otherShipDead = playerShip.attackShip(otherShip)
        
        if (otherShipDead) {
            retString += "The other ship explodes.\n"
            es.conclude()
            return retString
        }
        
        retString += "The other ship attacks you.\n"
        let playerShipDead = otherShip.attackShip(playerShip)
        
        if (playerShipDead) {
            retString += "Your ship explodes. You died!\n"
            es.conclude()
            return retString
        }
        
        return retString
    }
    
}
